packages/common/telemetry-constants.ts

```diff
@@ -1048,6 +1048,99 @@ export interface ReportsDatabaseGrafanaBannerClickedEvent {
   }
 }
 
+/**
+ * User clicked the deploy button for an Edge Function. This can happen in the Functions Editor page or through the functions ai assistant side panel.
+ *
+ * @group Events
+ * @source studio
+ * @page /dashboard/project/{ref}/functions/new
+ */
+export interface EdgeFunctionDeployButtonClickedEvent {
+  action: 'edge_function_deploy_button_clicked'
+  properties: {
+    origin: 'functions_editor' | 'functions_ai_assistant'
+  }
+  groups: {
+    project: string
+    organization: string
+  }
+}
+
+/**
+ * User clicked the AI Assistant button for creating an Edge Function.
+ * This can happen through clicking AI Assistant on the main block when there are no functions or in the secondary action section of the page or chat button in the functions editor.
+ *
+ * @group Events
+ * @source studio
+ * @page /dashboard/project/{ref}/functions
+ */
+export interface EdgeFunctionAiAssistantButtonClickedEvent {
+  action: 'edge_function_ai_assistant_button_clicked'
+  /**
+   *
+   */
+  properties: {
+    origin: 'no_functions_block' | 'secondary_action' | 'functions_editor_chat'
+  }
+  groups: {
+    project: string
+    organization: string
+  }
+}
```

Comment:
```suggestion
/**
 * User clicked the AI Assistant button to create an Edge Function.
 *
 * @group Events
 * @source studio
 * @page /dashboard/project/{ref}/functions
 */
export interface EdgeFunctionAiAssistantButtonClickedEvent {
  action: 'edge_function_ai_assistant_button_clicked'
  properties: {
    /**
      * Click on AI Assistant can either happen:
      *   1. on the main block when there are no functions
      *   2. in the secondary action section of the page
      *   3. on the chat button in the functions editor
      */
    origin: 'no_functions_block' | 'secondary_action' | 'functions_editor_chat'
  }
  groups: {
    project: string
    organization: string
  }
}
```

Just a suggestion to maybe add the documentation for origin directly next to the origin property